test/extensions/transport_sockets/tcp_stats/tcp_stats_test.cc

```diff
@@ -127,14 +127,14 @@ TEST_F(TcpStatsTest, SyscallFailureShortRead) {
   tcp_info_.tcpi_notsent_bytes = 42;
   EXPECT_CALL(io_handle_, getOption(IPPROTO_TCP, TCP_INFO, _, _))
       .WillOnce(Invoke([this](int, int, void* optval, socklen_t* optlen) {
-        *optlen = *optlen - 1;
-        memcpy(optval, &tcp_info_, sizeof(*optlen));
+        *optlen = offsetof(struct tcp_info, LAST_TCP_INFO_FIELD_WE_USE);
```

Comment:
Does this need a `- 1` to make this test work as intended?

Reply:
note that this is just the offset, not offset+sizeof(field). So, it's already a less value than we expect.